packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/DateTimeDescription.java

```diff
@@ -0,0 +1,213 @@
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ * Custom dateTime widget description.
+ *
+ * @author lfasani
+ */
+@Immutable
+public final class DateTimeDescription extends AbstractWidgetDescription {
+
+    private Function<VariableManager, String> idProvider;
+
+    private Function<VariableManager, String> labelProvider;
+
+    private Function<VariableManager, List<String>> iconURLProvider;
+
+    private Function<VariableManager, String> stringValueProvider;
+
+    private BiFunction<VariableManager, String, IStatus> newValueHandler;
+
+    private Function<VariableManager, DateTimeStyle> styleProvider;
+
+    private Function<VariableManager, Boolean> isDateOnlyProvider;
+
+    private DateTimeDescription() {
+        // Prevent instantiation
+    }
+
+    public static Builder newDateTimeDescription(String id) {
+        return new Builder(id);
+    }
+
+    public Function<VariableManager, String> getIdProvider() {
+        return this.idProvider;
+    }
+
+    public Function<VariableManager, String> getLabelProvider() {
+        return this.labelProvider;
+    }
+
+    public Function<VariableManager, List<String>> getIconURLProvider() {
+        return this.iconURLProvider;
+    }
+
+    public Function<VariableManager, String> getStringValueProvider() {
+        return this.stringValueProvider;
+    }
+
+    public Function<VariableManager, Boolean> isDateOnlyProvider() {
+        return this.isDateOnlyProvider;
+    }
+
+    public BiFunction<VariableManager, String, IStatus> getNewValueHandler() {
+        return this.newValueHandler;
+    }
+
+    public Function<VariableManager, DateTimeStyle> getStyleProvider() {
+        return this.styleProvider;
+    }
+
+    @Override
+    public String toString() {
+        String pattern = "{0} '{'idProvider: {1}, labelProvider: {2}, iconURLProvider: {3}, stringValueProvider: {4}, newValueHandler: {5}'}'";
+        return MessageFormat.format(pattern, this.getClass().getSimpleName(), this.getIdProvider(), this.getLabelProvider(), this.getIconURLProvider(), this.getStringValueProvider(), this.getNewValueHandler());
+    }
+
+    /**
+     * Builder used to create the SliderDescription.
+     */
+    @SuppressWarnings("checkstyle:HiddenField")
+    public static final class Builder {
+
+        private final String id;
+
+        private Function<VariableManager, String> idProvider;
+
+        private Function<VariableManager, String> targetObjectIdProvider;
+
+        private Function<VariableManager, String> labelProvider;
+
+        private Function<VariableManager, List<String>> iconURLProvider = variableManager -> List.of();
+
+        private Function<VariableManager, String> helpTextProvider;
+
+        private Function<VariableManager, Boolean> isReadOnlyProvider = variableManager -> false;
+
+        private Function<VariableManager, String> stringValueProvider;
+
+        private Function<VariableManager, Boolean> isDateOnlyProvider;
```

Comment:
You should probably add a default value here

Reply:
Replaced by an enum